app/uk/gov/hmrc/economiccrimelevyregistration/services/RegistrationValidationService.scala

```diff
@@ -326,4 +337,59 @@ class RegistrationValidationService @Inject() (clock: Clock, schemaValidator: Sc
 
   private def missingErrorMessage(missingDataDescription: String): String = s"$missingDataDescription is missing"
 
+  private def validateOtherEntity(
+    registration: Registration
+  ): ValidationResult[Either[EclSubscription, Registration]] =
+    (
+      validateAmlSupervisor(registration),
+      validateOptExists(registration.businessSector, "Business sector"),
+      validateContactDetails("First", registration.contacts.firstContactDetails),
+      validateSecondContactDetails(registration.contacts),
+      validateAmlRegulatedActivity(registration),
+      validateOptExists(registration.relevantAp12Months, "Relevant AP 12 months choice"),
+      validateOptExists(registration.relevantApRevenue, "Relevant AP revenue"),
+      validateConditionalOptExists(
+        registration.relevantApLength,
+        registration.relevantAp12Months.contains(false),
+        "Relevant AP length"
+      ),
+      validateRevenueMeetsThreshold(registration),
+      validateOptExists(registration.optOtherEntityJourneyData, "other entity data"),
+      validateOptExists(registration.otherEntityJourneyData.businessName, "business name"),
+      registration.otherEntityJourneyData.entityType match {
+        case Some(Charity) => validateCharity(registration)
+        case _             => ???
+      }
```

Comment:
```suggestion
      validateOptExists(registration.optOtherEntityJourneyData, "Other entity data"),
      validateOptExists(registration.otherEntityJourneyData.businessName, "Business name"),
      registration.otherEntityJourneyData.entityType match {
        case Some(Charity) => validateCharity(registration)
        case _             => ???
      }
```

Capitalisation for consistency and also there are tests missing for if the other entity data is invalid. See the similar test for regular entity type - it can be similar to this

Reply:
Added tests.